### PR TITLE
Vary page layout for registering new accounts

### PR DESCRIPTION
**Pull request by bolatovumar**
address #1138

### CODE REVIEW

File: BTCPayServer/Controllers/AccountController.cs [hunk 1]
```diff
@@ -369,6 +369,7 @@ public async Task<IActionResult> Register(string returnUrl = null, bool logon =
             var policies = await _SettingsRepository.GetSettingAsync<PoliciesSettings>() ?? new PoliciesSettings();
             if (policies.LockSubscription && !User.IsInRole(Roles.ServerAdmin))
                 return RedirectToAction(nameof(HomeController.Index), "Home");
+            ViewData["WithReturnUrl"] = returnUrl != null ? true : false;
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/AccountController.cs]: I don't think you should use `returnUrl`. There can be a returnUrl if you try to go to a page that is forbidden, you get redirected to login page and then if you click on "Create an account" in the login page, the returnUrl get carried out.

File: BTCPayServer/Views/Account/Register.cshtml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ﻿@model RegisterViewModel
 @{
     ViewData["Title"] = "Register";
-    Layout = "_WelcomeLayout.cshtml";
+    Layout = (bool)ViewData["WithReturnUrl"] ? "../Shared/_Layout.cshtml" : "_WelcomeLayout.cshtml";
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Account/Register.cshtml]: You should use Another ViewData that is only set in `Add new user` for the reason I specified above.

File: BTCPayServer/Views/Account/Register.cshtml [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 }
 
-<div class="modal-dialog modal-login">
+<div class="modal-dialog @(Layout != "_WelcomeLayout.cshtml" ? "modal-dialog-centered" : "")">
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Account/Register.cshtml]: this is ugly to read. I advise you do to on the top block.

```
var isDisruptive = ViewData["DisruptiveLayout"] is bool;
Layout = isDisruptive ? "_WelcomeLayout.cshtml" : "../Shared/_Layout.cshtml";
```

then use` isDisruptive` in your condition here.

File: BTCPayServer/Views/Account/Register.cshtml (hunk 2 above)
Comment (bolatovumar) [reply BTCPayServer/Views/Account/Register.cshtml]: Makes sense.

File: BTCPayServer/Views/Account/Register.cshtml (hunk 3 above)
Comment (bolatovumar) [reply BTCPayServer/Views/Account/Register.cshtml]: Makes sense.

File: BTCPayServer/Views/Account/Register.cshtml (hunk 2 above)
Comment (NicolasDorier) [reply BTCPayServer/Views/Account/Register.cshtml]: no, I am saying that `ViewData["BoringLayout"] = true;` should be set in the ListUser's add new user.

Comment (bolatovumar) [reply BTCPayServer/Views/Account/Register.cshtml]: Oh, ok.

**Comment by NicolasDorier** [comment]
way simpler! @pavlenex you take  a look?

**Comment by pavlenex** [comment]
tACK.

**Pull request merged**